### ISSUE TITLE
Anyone have an interest in making ReduceLROnPlateau compatible with weight decay optimizers?

### ISSUE THREAD

**Issue by ben-arnao** [tensorflow/tensorflow#37358]
**System information**
- TensorFlow version (you are using): 2.0, should work with 2.1 too
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

**Will this change the current api? How?** No

**Who will benefit with this feature?** Those using decoupled weight decay optimizers with reduce LR callback

**Any Other info.**

It has been shown that decoupling weight decay from the learning rate can simply hyper parameter search and leads to better performance.

https://arxiv.org/abs/1711.05101

Tensorflow has implementations for Adam and SGD.

https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/SGDW

However the docs mention that when we decay the lr, we also need to decay the weight decay itself by the same factor. This is easy to do if we are using our own static schedulers like in the example they provided but i think reduce LR on plateau is still a valid technique and adding the weight decay-decay into this callback is fairly straightforward.

All we would really need to do is check if there is weight decay and if there is, just decay it just the same as with the lr. Existing functionality would remain unchanged and if there is weight decay, it should automatically be decayed anyway with the lr right? even if that is not the case, a simple additional argument "decay_wd=False" should suffice.

```
      if self.monitor_op(current, self.best):
        self.best = current
        self.wait = 0
      elif not self.in_cooldown():
        self.wait += 1
        if self.wait >= self.patience:
          old_lr = float(K.get_value(self.model.optimizer.lr))
          old_wd = float(K.get_value(self.model.optimizer.weight_decay))
          if old_lr > self.min_lr:
            new_lr = old_lr * self.factor
            new_wd = old_wd * self.factor
            new_lr = max(new_lr, self.min_lr)
            K.set_value(self.model.optimizer.weight_decay, new_wd)
            K.set_value(self.model.optimizer.lr, new_lr)
            if self.verbose > 0:
              print('\nEpoch %05d: ReduceLROnPlateau reducing learning '
                    'rate to %s.' % (epoch + 1, new_lr))
            self.cooldown_counter = self.cooldown
            self.wait = 0
```

**Comment by gowthamkpr** [comment]
@ben-arnao Please post this issue [here](https://github.com/tensorflow/addons/issues) as it is more appropriate to the tensorflow addons repo than tensorflow itself as this has been implemented in addons as shown [here](https://github.com/tensorflow/addons/blob/1af92905ed03f05fcf6f4918783c3d151a8b8350/tensorflow_addons/optimizers/weight_decay_optimizers.py#L266-L336).
